src/storage/sqlite.rs

```diff
@@ -58,12 +58,101 @@ pub struct SqliteStorage {
 }
 
 impl SqliteStorage {
+    fn add_bool(
+        obj: &mut Object,
+        atype: CK_ATTRIBUTE_TYPE,
+        val: ValueRef,
+    ) -> Result<()> {
+        obj.set_attr(match val.as_i64_or_null().map_err(bad_storage)? {
+            Some(b) => Attribute::from_bool(atype, b != 0),
+            None => return Err(CKR_ATTRIBUTE_VALUE_INVALID)?,
+        })
+    }
+
+    fn add_num(
+        obj: &mut Object,
+        atype: CK_ATTRIBUTE_TYPE,
+        val: ValueRef,
+    ) -> Result<()> {
+        obj.set_attr(match val.as_i64_or_null().map_err(bad_storage)? {
+            Some(n) => Attribute::from_ulong(atype, Self::val_to_ulong(n)?),
+            None => return Err(CKR_ATTRIBUTE_VALUE_INVALID)?,
+        })
+    }
+
+    fn add_string(
+        obj: &mut Object,
+        atype: CK_ATTRIBUTE_TYPE,
+        val: ValueRef,
+    ) -> Result<()> {
+        obj.set_attr(match val.as_str_or_null().map_err(bad_storage)? {
+            Some(s) => Attribute::from_string(atype, s.to_string()),
+            None => return Err(CKR_ATTRIBUTE_VALUE_INVALID)?,
+        })
+    }
+
+    fn add_byte_array(
+        obj: &mut Object,
+        atype: CK_ATTRIBUTE_TYPE,
+        val: ValueRef,
+    ) -> Result<()> {
+        obj.set_attr(match val.as_blob_or_null().map_err(bad_storage)? {
+            Some(v) => Attribute::from_bytes(atype, v.to_vec()),
+            None => Attribute::from_bytes(atype, Vec::new()),
+        })
+    }
+
+    fn add_ulong_array(
+        obj: &mut Object,
+        atype: CK_ATTRIBUTE_TYPE,
+        val: ValueRef,
+    ) -> Result<()> {
+        obj.set_attr(match val.as_blob_or_null().map_err(bad_storage)? {
+            Some(blob) => {
+                let ulen = std::mem::size_of::<u64>();
```

Comment:
`sizeof!()`

Reply:
does it really make sense to have a macro just to shorten this a bit?

Would you be ok if I import std::mem::size_of and then just call `size_of::<u64>()` in all these places?

Reply:
Probably not. I missed the conversion in there. Lets ignore these comments.